### ISSUE TITLE
There is miscaculated attention mask shape in comment in GRPO trainer.

### ISSUE THREAD

**Issue by linkedlist771** [huggingface/trl#2924]
### Reproduction

There is no need to have a python code for it is the comment's problem.
In the code https://github.com/huggingface/trl/blob/main/trl/trainer/grpo_trainer.py#L742, its comment says the shape of the attention mask should be:
```python
# (B*G, P+C)
```
However, the attention mask here is the mask for the prompt tokens and the completion tokens for this `process device`, that means it only generate one sample for each batch. It should be `# (B, P+C)`,  I have logger out the attention musk, which also validates this:
``` bash
attention_mask: tensor([[1, 1, 1,  ..., 1, 1, 1],
        [1, 1, 1,  ..., 0, 0, 0],
        [1, 1, 1,  ..., 0, 0, 0],
        ...,
        [1, 1, 1,  ..., 0, 0, 0],
        [1, 1, 1,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0]], device='cuda:0'), shape: torch.Size([8, 618])
```
I have sumitted a [PR2925](https://github.com/huggingface/trl/pull/2925)  for this.

### System Info

No need for system info.

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete
